tsdb/engine/tsm1/engine.go

```diff
@@ -880,7 +880,9 @@ func (e *Engine) CreateIterator(opt influxql.IteratorOptions) (influxql.Iterator
 }
 
 // createVarRefIterator creates an iterator for a variable reference.
-func (e *Engine) createVarRefIterator(opt influxql.IteratorOptions) ([]influxql.Iterator, error) {
+// The limit argument determines if the limit optimization added for #6391 is used.
+// Limit optimization is disabled for aggregate queries temporarily. See #6661.
+func (e *Engine) createVarRefIterator(opt influxql.IteratorOptions, limit bool) ([]influxql.Iterator, error) {
```

Comment:
It seems like it'd be more useful in the future if this was named `aggregate` to signal that it's for an aggregate function.